File: 45-jump-game-ii/45-jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        prev_max = 0
        cur_max = 0
        count = 0
        for i, num in enumerate(nums):
            if i > prev_max:
                prev_max = cur_max
                count += 1
            cur_max = max(cur_max, num + i)
        return count
        

File: 45-jump-game-ii/45-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int stepCount = 0;
        int curFarthest = 0;
        int jumpEnd = 0;
        for(int i = 0; i < nums.length ; i++){
            curFarthest = Math.max(curFarthest, nums[i] + i);
            if(jumpEnd == i){
                if (jumpEnd == nums.length - 1)return stepCount;
                stepCount += 1;
                jumpEnd = curFarthest;
            }
        }
        return stepCount;
    }
}